src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1872,21 +1872,34 @@ internal override CommonMessageProvider MessageProvider
         /// <summary>
         /// The bag in which semantic analysis should deposit its diagnostics.
         /// </summary>
-        internal DiagnosticBag SemanticDiagnostics
+        internal DiagnosticBag DeclarationDiagnostics
         {
             get
             {
-                if (_lazySemanticDiagnostics == null)
+                // We should only be placing diagnostics in this bag until
+                // we are done gathering declaration diagnostics. Assert that is
+                // the case. But since we have bugs (see https://github.com/dotnet/roslyn/issues/846)
+                // we disable the assertion until they are fixed.
```

Comment:
No TODO comment. In fact, we will soon embark on the task of cleaning out the TODOs and adding github issues to replace them.